### PR TITLE
Track extensions

### PR DESCRIPTION
**Pull request by bsathian**
http://uaf-10.t2.ucsd.edu/~bsathian/SDL_plots/20211210-PullRequest/eff_plots__GPU_unified_efcd8c0_PU200/mtv/

http://uaf-10.t2.ucsd.edu/~bsathian/SDL_plots/20211210-PullRequest/eff_plots__GPU_unified_efcd8c0_PU200/mtv/TC_v_TCE_AllTypes__pt.pdf

### CODE REVIEW

**Comment by bsathian** [comment]
http://uaf-10.t2.ucsd.edu/~bsathian/SDL_plots/20211210-PullRequest/eff_plots__GPU_unified_efcd8c0_PU200/mtv/

File: SDL/Kernels.cuh [hunk 1]
```diff
@@ -40,30 +41,19 @@ const unsigned int N_MAX_SEGMENTS_PER_MODULE = 600; //Also used in (pixel)trackl
 const unsigned int N_MAX_PIXEL_MD_PER_MODULES = 100000;
 const unsigned int N_MAX_PIXEL_SEGMENTS_PER_MODULE = 50000;
 
-//const unsigned int N_MAX_TRIPLETS_PER_MODULE = 5000;
```
Comment (tresreid) [inline SDL/Kernels.cuh:43]: Can you keep these in the comments. It might still be useful if we need to increase these values.

**Comment by bsathian** [comment]
To-do
1. Keep comments
2. Make compatible with streams
3. Togglable extensions

**Comment by bsathian** [comment]
Updates

1. Track Extensions properly hidden behind #ifdefs. Using -e toggles track extensions, and using -3 toggles track extensions and T3T3 extensions
2. Made compatible with streams (though only 2 streams can be run with T3T3 enabled)
3. The old commented lines with the limits have been restored

**Comment by tresreid** [comment]
The validation doesn't produce plots.
The explicit version uses cudaMallocManaged but that should be entirely cudaMallocs

**Comment by tresreid** [comment]
also the caching versions do not work at all

**Comment by bsathian** [comment]
Fixed things

1. Cache mode runs now
2. cudaMallocManaged instances have been removed in the explicit case
3. Validation produces plots (http://uaf-10.t2.ucsd.edu/~bsathian/Track_extension_PR/)

**Comment by tresreid** [comment]
unified cache version is missing 10 cudeFree statements (other versions seem fine)

**Comment by tresreid** [comment]
validation plots still don't function as intended. in "sdl_validate_efficiency" you need to add the -e flags in the lines inf run_gpu() and in the runs at the bottom of the page (or better yet, make an easy switch to change them both when needed". in "sdl_plot_efficiency" you need to change "f:e:3:h" to "fe3h" (see the review comment).

**Comment by tresreid** [comment]
Timing and validation setup will be fixed in a patch

**Pull request merged**